### PR TITLE
feat: return text when content type is text/csv

### LINKED ISSUE #578: Using googleapis for direct CSV downloads does not return CSV
When googleapis client library is update gaxios to 6.0.0 the CSV media download stopped working

```
const { data } = await dfareporting.reports.files.get({
  profileId,
  reportId,
  fileId,
  alt: 'media',
});
```

before 6.0.0, `data` would be the CSV string, after it would be an empty object

#### Environment details

  - OS: Linux
  - Node.js version: 18.18.1
  - npm version: 9.8.1
  - `gaxios` version: 6.0.0

#### Steps to reproduce

  1. Use googleapis with media download for CSV (eg. `dfareporting` GET `/userprofiles/4455035/reports/500062091/files/2000369329?alt=media`
  2.  Response body is empty rather than giving the the CSV file

### PR DESCRIPTION
**Pull request by vinteo**
Thank you for opening a Pull Request! Before submitting your PR, there are a few things you can do to make sure it goes smoothly:
- [ ] Make sure to open an issue as a [bug/issue](https://github.com/googleapis/gaxios/issues/new/choose) before writing your code!  That way we can discuss the change, evaluate designs, and agree on the general idea
- [ ] Ensure the tests and linter pass
- [ ] Code coverage does not decrease (if any source code was changed)
- [ ] Appropriate docs were updated (if necessary)

Fixes #578 🦕

### CODE REVIEW

File: src/gaxios.ts [hunk 1]
```diff
@@ -404,6 +404,7 @@ export class Gaxios {
       return data as {};
     } else if (
       contentType.includes('text/plain') ||
+      contentType.includes('text/csv') ||
```
Comment (alexander-fenster) [inline src/gaxios.ts:407]: Now I'm just thinking if we should instead do `contentType.startsWith('text/')` and consider all text content the same. @sofisl what do you think?

**Review by alexander-fenster — APPROVED** [review]
Thank you! We'll merge it and release soon.

File: src/gaxios.ts (hunk 1 above)
Comment (sofisl) [reply src/gaxios.ts:407]: I agree! Would you mind including a regex instead? and maybe a `.match`? Thank you!

**Review by sofisl — CHANGES_REQUESTED** [review]

File: src/gaxios.ts [hunk 2]
```diff
@@ -402,10 +402,7 @@ export class Gaxios {
         // continue
       }
       return data as {};
-    } else if (
-      contentType.includes('text/plain') ||
-      contentType.includes('text/html')
-    ) {
+    } else if (contentType.match(/^text\//)) {
```
Comment (vinteo) [inline src/gaxios.ts:405]: not sure how specific the regular expression should be, hope this suits.

**Review by danielbankhead — APPROVED** [review]

**Comment by vinteo** [comment]
Any chance of pushing this through?

**Review by sofisl — APPROVED** [review]

**Pull request merged**